packages/stargate/src/accounts.ts

```diff
@@ -40,49 +40,103 @@ function accountFromBaseAccount(input: BaseAccount): Account {
-      assert(baseAccount);
-      return accountFromBaseAccount(baseAccount);
-    }
-    case "/cosmos.vesting.v1beta1.DelayedVestingAccount": {
-      const baseAccount = DelayedVestingAccount.decode(value)?.baseVestingAccount?.baseAccount;
-      assert(baseAccount);
-      return accountFromBaseAccount(baseAccount);
-    }
-    case "/cosmos.vesting.v1beta1.PeriodicVestingAccount": {
-      const baseAccount = PeriodicVestingAccount.decode(value)?.baseVestingAccount?.baseAccount;
-      assert(baseAccount);
-      return accountFromBaseAccount(baseAccount);
-    }
+  const parseModuleAccount: AccountParser = ({ value }: Any): Account => {
+    const baseAccount = ModuleAccount.decode(value).baseAccount;
+    assert(baseAccount);
+    return accountFromBaseAccount(baseAccount);
+  };
 
-    default:
-      throw new Error(`Unsupported type: '${typeUrl}'`);
-  }
+  const parseBaseVestingAccount: AccountParser = ({ value }: Any): Account => {
+    const baseAccount = BaseVestingAccount.decode(value)?.baseAccount;
+    assert(baseAccount);
+    return accountFromBaseAccount(baseAccount);
+  };
+
+  const parseContinuousVestingAccount: AccountParser = ({ value }: Any): Account => {
+    const baseAccount = ContinuousVestingAccount.decode(value)?.baseVestingAccount?.baseAccount;
+    assert(baseAccount);
+    return accountFromBaseAccount(baseAccount);
+  };
+
+  const parseDelayedVestingAccount: AccountParser = ({ value }: Any): Account => {
+    const baseAccount = DelayedVestingAccount.decode(value)?.baseVestingAccount?.baseAccount;
+    assert(baseAccount);
+    return accountFromBaseAccount(baseAccount);
+  };
+
+  const parsePeriodicVestingAccount: AccountParser = ({ value }: Any): Account => {
+    const baseAccount = PeriodicVestingAccount.decode(value)?.baseVestingAccount?.baseAccount;
+    assert(baseAccount);
+    return accountFromBaseAccount(baseAccount);
+  };
+
+  return new Map<Any["typeUrl"], AccountParser>([
+    ["/cosmos.auth.v1beta1.BaseAccount", parseBaseAccount],
+    ["/cosmos.auth.v1beta1.ModuleAccount", parseModuleAccount],
+    ["/cosmos.vesting.v1beta1.BaseVestingAccount", parseBaseVestingAccount],
+    ["/cosmos.vesting.v1beta1.ContinuousVestingAccount", parseContinuousVestingAccount],
+    ["/cosmos.vesting.v1beta1.DelayedVestingAccount", parseDelayedVestingAccount],
+    ["/cosmos.vesting.v1beta1.PeriodicVestingAccount", parsePeriodicVestingAccount],
+  ]);
+}
+
+/**
+ * Basic implementation of AccountParser. This is supposed to support the most relevant
+ * common Cosmos SDK account types. If you need support for additional account types,
+ * you'll need to use `AccountParserManager` and `createAccountParserRegistry` directly.
+ *
+ * @example
+ * ```
+ * const myAccountParserManager = new AccountParserManager(createAccountParserRegistry());
+ * myAccountParserManager.register('/custom.type.v1.CustomAccount', customAccountParser);
+ * const account = customParserManager.parseAccount(someInput);
```

Comment:
```suggestion
 * const { register, parseAccount } = new AccountParserManager(createAccountParserRegistry());
 * register('/custom.type.v1.CustomAccount', customAccountParser);
 * const options: StargateClientOptions = { accountParser: parseAccount };
```

This probably aligns closer to expected usage